portfolio/src/main/webapp/script.js

```diff
@@ -58,11 +58,16 @@ function handleResponse(response) {
 function addTextToDom(text) {
     console.log("Adding text to DOM: "+text);
     const table = document.getElementById("comment-section");
-    const parsedText = JSON.parse(text);
+    const parsedText = JSON.parse(text); // this is the object with properties numComments and comments
+    parsedText.comments = JSON.parse(parsedText.comments);
+    sessionStorage["size"] = parsedText.comments.length;
 
     var i;
-    for (i = 0; i < parsedText.length; i++) {
-        const comment = parsedText[i];
+    const currPage = sessionStorage["current-page"];
+    const startComment = (currPage - 1) * sessionStorage["num-comments"];
+    const commentsToShow = currPage * sessionStorage["num-comments"];
+    for (i = startComment; i < Math.min(sessionStorage["size"], commentsToShow); i++) {
+        const comment = parsedText.comments[i];
         const heading = ` (posted at ${comment.date})`;
         const row = makeComment(comment.name + heading, comment.content);
         table.appendChild(row);
```

Comment:
Do you need to empty the table before appending to it or is it always empty when you reach this line of code?

Reply:
it's always empty